src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1569,6 +1633,15 @@ private void InitializeNodeLibrary()
             }
 
             CustomNodeManager.AddUninitializedCustomNodesInPath(pathManager.CommonDefinitions, IsTestMode);
+
+            //Initialize searcher, if the applyAllDeletes is true all buffered deletes on documents will be applied (made visible) in the returned reader
+            dirReader = writer.GetReader(applyAllDeletes: true);
+            IndexSearcher searcher = new IndexSearcher(dirReader);
+
+            SearchModel.Searcher = searcher;
+
+            writer.Commit();
```

Comment:
This is the actual spot where we write index to disk right? We should guard against this under test mode not the initialization I think